src/Symfony/Component/Console/Output/StreamOutput.php

```diff
@@ -70,7 +70,11 @@ public function getStream()
      */
     protected function doWrite($message, $newline)
     {
-        if (false === @fwrite($this->stream, $message) || ($newline && (false === @fwrite($this->stream, PHP_EOL)))) {
+        if ($newline) {
+            $message .= PHP_EOL;
```

Comment:
Not reassigning was done purposely to save memory (https://github.com/symfony/symfony/pull/18932)

Reply:
well, at least, this code avoids the concatenation when not adding a newline (while all writes where doing a concatenation before #18932 and I'm not sure PHP optimizes the concatenation of empty strings).

But I fear we cannot avoid both the extra allocation and the race condition (you could then ensure that your put the newline in your big message instead of asking the console component to add it though)

Reply:
Fine for me.

> you could then ensure that your put the newline in your big message instead of asking the console component to add it though

Perhaps this should be mentioned somewhere (docblock and/or symfony-docs?)